apps/examples/safe_test/safe_test_receiver.c

```diff
@@ -0,0 +1,236 @@
+ * apps/examples/safe_test/safe_receiver.c
+ *
+ *   Copyright (C) 2008, 2011-2012 Gregory Nutt. All rights reserved.
+ *   Author: Gregory Nutt <gnutt@nuttx.org>
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in
+ *    the documentation and/or other materials provided with the
+ *    distribution.
+ * 3. Neither the name NuttX nor the names of its contributors may be
+ *    used to endorse or promote products derived from this software
+ *    without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS
+ * OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+ * AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <mqueue.h>
+#include <signal.h>
+#include <sys/types.h>
+
+#include "safe_test.h"
+
+static int sig_mq_num;
+static int mq_num;
+
+static int safe_test_sig_timedwait(int argc, char *argv[])
+{
+	int pid;
+	sigset_t sig;
+	siginfo_t info;
+	struct timespec timeout;
+	
+	pid = getpid();
+	printf("[%d] safe_test_sig_timedwait_task\n", pid);
+
+	timeout.tv_sec  = 0;
+	timeout.tv_nsec = SAFETEST_USLEEP;
+	
+	sigemptyset(&sig);
+	sigaddset(&sig, SAFETEST_SIG_TIME_WAIT_NO);
+	
+	while (1) {
+		sigtimedwait(&sig, &info, &timeout);
+		printf("[%d] sig timed wait\n", pid);
+	}
+	return 0;
+}
+
+static int safe_test_sem_timedwait(int argc, char *argv[])
+{
+	int pid;
+	struct timespec abstime;
+	sem_t sem;
+
+	pid = getpid();
+	printf("[%d] safe_test_sem_timedwait_task\n", pid);
+	sem_init(&sem, 0, 0);
+	
+	abstime.tv_sec = 0;
+	abstime.tv_nsec = SAFETEST_USLEEP;
+	
+	while (1) {
+		sem_timedwait(&sem, &abstime);
```

Comment:
Is it okay not to check the return of timedwait?
Because it is sometimes broken with signal.